ws/register.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2020 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+package ws
+
+import (
+	"errors"
+)
+
+var (
+	ErrUnsupportedTypeError = errors.New("unsupported type")
+	ErrUnregisteredProtocol = errors.New("protocol not registered")
+)
+
+type Protocol interface {
+	DeEncapsulate(m *ProtoMsg) (interface{}, error)
+	Encapsulate(m interface{}) (*ProtoMsg, error)
+}
+
+var protoMap = map[ProtoType]Protocol{}
```

Comment:
I don't understand the purpose of this interface. All messages holding `shell` data should be of `ProtoTypeShell`, thus this map holds at most one value.